src/api/app/models/project.rb

```diff
@@ -1587,8 +1587,8 @@ def self.validate_repository_xml_attribute(request_data, project_name)
   end
 
   def get_removed_repositories(request_data)
-    new_repositories = request_data.elements('repository').map(&:values).flatten
-    old_repositories = repositories.all.map(&:name)
+    new_repositories = request_data.elements('repository').map!(&:values).flatten
+    old_repositories = repositories.map(&:name)
```

Comment:
why not `!` here?

Reply:
This actually sounds like a job for pluck

Reply:
But to answer your question:
```
undefined method `map!' for #<Repository::ActiveRecord_Associations_CollectionProxy:0x00000000085d93c0>